cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/lineage/field/FieldLineageInfo.java

```diff
@@ -375,10 +376,10 @@ private void computeIncomingSummaryHelper(EndPointField destination, Operation c
           continue;
         }
 
-        Map<EndPointField, Set<EndPointField>> result = new HashMap<>();
         visitedOperationNames.add(write.getName());
         computeIncomingSummaryHelper(new EndPointField(write.getDestination(), input.getName()),
-                operationsMap.get(input.getOrigin()), write, visitedOperationNames, result);
+                                     operationsMap.get(input.getOrigin()), write, visitedOperationNames,
+                                     new HashMap<>());
```

Comment:
what I meant by my previous comment is that this parameter is not needed. The method computeIncomingSummaryHelper() only adds entries to it, but nobody ever reads from it. Or am I missing something?

Reply:
`computeIncomingSummaryHelper()` also gets called while computing incoming summary from line 288 above, which uses the HashMap. This call just uses the same helper function to reuse the recursion logic.

Reply:
Ah ok I did not see that in the diff. All good.